internal/jwtverify/token_verifier_jwt.go

```diff
@@ -10,34 +10,54 @@ import (
 	"time"
 
 	"github.com/cristalhq/jwt/v3"
+	"github.com/s12v/go-jwks"
 )
 
 type VerifierConfig struct {
 	// HMACSecretKey is a secret key used to validate connection and subscription
 	// tokens generated using HMAC. Zero value means that HMAC tokens won't be allowed.
 	HMACSecretKey string
+
 	// RSAPublicKey is a public key used to validate connection and subscription
 	// tokens generated using RSA. Zero value means that RSA tokens won't be allowed.
 	RSAPublicKey *rsa.PublicKey
+
+	// JWKSPublicURL is a public url used to validate connection and subscription
+	// tokens generated using rotating RSA public keys. Zero value means that RSA tokens won't be allowed.
+	JWKSPublicURL string
 }
 
 func NewTokenVerifierJWT(config VerifierConfig) *VerifierJWT {
 	verifier := &VerifierJWT{}
+
 	algorithms, err := newAlgorithms(config.HMACSecretKey, config.RSAPublicKey)
 	if err != nil {
 		panic(err)
 	}
 	verifier.algorithms = algorithms
+
+	if config.JWKSPublicURL != "" {
+		verifier.jwksClient = &jwksClient{jwks.NewDefaultClient(
+			jwks.NewWebSource(config.JWKSPublicURL),
```

Comment:
If write own http client which options have to be configured from config.json? Any hint?

Reply:
Still trying to realize what would be the best way here.

In general we don't need any extra configurations exposed except one you already added (`token_jwks_public_endpoint`). All other options can be exposed in future commits if needed.

I am trying to understand a purpose of cache in this situation. I understand that it's fine to expire some entries with time. But I don't understand why we need to periodically refresh once loaded keys - can concrete `kid` be changed? 

Even if we need to refresh: why not have much simpler cache – key is loaded once and has `ttl`, after `ttl` passed it is just removed from cache. So subsequent request will get it from jwks endpoint again. Maybe it's a good idea to load key using https://godoc.org/golang.org/x/sync/singleflight package to avoid spikes in load on jwks_endpoint. But I don't understand the purpose of 2 different durations in `s12v/go-jwks` at the moment. 

@danikarik maybe you understand more details in key update process and have some thoughts?